File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        Deque<TreeNode> deq = new ArrayDeque();
        List<Integer> rs = new ArrayList();
        if(root != null){
            deq.addLast(root);
        }
        while(!deq.isEmpty()){
            int size = deq.size();
            TreeNode curr =null;
            for(int i = 0; i < size; i++){
                curr = deq.removeFirst();
                if(curr.left != null){
                    deq.addLast(curr.left);
                }
                if(curr.right != null){
                    deq.addLast(curr.right);
                }
            }
            rs.add(curr.val);
        }
        return rs;
    }
}

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func isBalanced(root *TreeNode) bool {
    if root == nil {
		return true
	}

	childrenBalanced := isBalanced(root.Left) && isBalanced(root.Right)

	limbDif := absolute(maxDepth(root.Left) - maxDepth(root.Right))

	return childrenBalanced && limbDif<2

}

func maxDepth(root *TreeNode) int {
	if root == nil {
		return 0
	}

	max := maxInt(maxDepth(root.Left),maxDepth(root.Right))

	return max + 1
}

func absolute(c int) int {

	if c > 0 {
		return c
	}
	return c*-1
}


func maxInt(a,b int) int{
	if a>b{
		return a
	}
	return b
}
